sklearn/utils/sparsefuncs.py

```diff
@@ -341,6 +341,11 @@ def inplace_swap_column(X, m, n):
 
 def _minor_reduce(X, ufunc):
     major_index = np.flatnonzero(np.diff(X.indptr))
+
+    # reduceat tries casts X.indptr to intp, which errors
+    # if it is int64 on a 32 bit system.
+    # Reinitializing prevents this where possible, see #13737
+    X = type(X)((X.data, X.indices, X.indptr), shape=X.shape)
```

Comment:
so what does this reinitialisation actually do? Will it downcast indptr if X.shape is less than int32.max???

Reply:
Yes, precisely. The logic is in `get_index_dtype` [here](https://github.com/scipy/scipy/blob/master/scipy/sparse/sputils.py).

Reply:
And in other cases is this a non-copying operation?

Reply:
Yes (see `__init__` for `_cs_matrix` [here](https://github.com/scipy/scipy/blob/master/scipy/sparse/compressed.py)). That could be made explicit by adding `copy=False`.